mumd/src/network/tcp.rs

```diff
@@ -225,19 +225,21 @@ async fn send_voice(
             |phase| !matches!(phase, StatePhase::Connected(VoiceStreamType::TCP)),
             async {
                 loop {
-                    packet_sender.send(
+                    let res: Result<(), ServerSendError> = packet_sender.send(
                         receiver
                             .lock()
                             .await
                             .next()
                             .await
-                            .unwrap() //TODO handle panic
-                            .into())
-                        .unwrap(); //TODO handle panic
+                            .expect("No audio stream")
+                            .into());
+                    if matches!(res, Err(_)) {
+                        return res;
+                    }
```

Comment:
This could be written with the question mark operator instead.

Reply:
Unless you can turbofish `?`, no.

```
diff --git a/mumd/src/network/tcp.rs b/mumd/src/network/tcp.rs
index 5783cc8..88bc76a 100644
--- a/mumd/src/network/tcp.rs
+++ b/mumd/src/network/tcp.rs
@@ -225,17 +225,14 @@ async fn send_voice(
             |phase| !matches!(phase, StatePhase::Connected(VoiceStreamType::TCP)),
             async {
                 loop {
-                    let res: Result<(), ServerSendError> = packet_sender.send(
+                    packet_sender.send(
                         receiver
                             .lock()
                             .await
                             .next()
                             .await
                             .expect("No audio stream")
-                            .into());
-                    if matches!(res, Err(_)) {
-                        return res;
-                    }
+                            .into())?;
                 }
             },
             inner_phase_watcher.clone(),
```

```
    Checking mumd v0.3.0 (/home/gustav/dev/mum/mumd)
warning: unused import: `ServerSendError`
 --> mumd/src/network/tcp.rs:1:20
  |
1 | use crate::error::{ServerSendError, TcpError};
  |                    ^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0698]: type inside `async fn` body must be known in this context
   --> mumd/src/network/tcp.rs:239:27
    |
239 |         ).await.unwrap_or(Ok(()))?;
    |                           ^^ cannot infer type for type parameter `E` declared on the enum `Result`
    |
note: the type is part of the `async fn` body because of this `await`
   --> mumd/src/network/tcp.rs:224:9
    |
224 | /         run_until(
225 | |             |phase| !matches!(phase, StatePhase::Connected(VoiceStreamType::TCP)),
226 | |             async {
227 | |                 loop {
...   |
238 | |             inner_phase_watcher.clone(),
239 | |         ).await.unwrap_or(Ok(()))?;
    | |_______________^

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0698`.
error: could not compile `mumd`

To learn more, run the command again with --verbose.
```